### PR TITLE
fixes the jumpsuit code made by some idiot

### PR DESCRIPTION
**Pull request by Sylphily**
## About The Pull Request
fixes code not actually uncovering the chest area when it should of, and this was fucking done by moving code related to jumpsuits to Under.dm, instead of Clothing.dm, i dont know why this fixes it but it does, why it wasn't put there to be with i am more confused about.

## Why It's Good For The Game
this is intended, but the dumbass didn't put it under the jumpsuit code

## Changelog
:cl:Shotgun.boom#2723
fix: jumpsuits actually uncovers the chest
code: moved jumpsuit code from clothing.dm to under.dm
/:cl:

### CODE REVIEW

**Comment by Sylphily** [comment]
what the fuck what is conflicting

**Comment by Crossedfall** [comment]
> what the fuck what is conflicting

image

Pull the latest master and verify your changes.

File: code/modules/clothing/under/_under.dm [hunk 1]
```diff
@@ -173,3 +173,98 @@
...
+/obj/item/clothing/under/verb/toggle()
+	set name = "Adjust Suit Sensors"
+	set category = "Object"
+	set src in usr
+	var/mob/M = usr
+	if (istype(M, /mob/dead/))
+		return
+	if (!can_use(M))
+		return
+	if(src.has_sensor == LOCKED_SENSORS)
```
Comment (Archanial) [inline code/modules/clothing/under/_under.dm]:
```suggestion
	if(has_sensor == LOCKED_SENSORS)
```

File: code/modules/clothing/under/_under.dm [hunk 2]
```diff
@@ -173,3 +173,98 @@
...
+		to_chat(usr, "The controls are locked.")
+		return 0
```
Comment (Archanial) [inline code/modules/clothing/under/_under.dm]:
```suggestion
		return FALSE
```

File: code/modules/clothing/under/_under.dm [hunk 3]
```diff
@@ -173,3 +173,98 @@
...
+	if(src.has_sensor == BROKEN_SENSORS)
```
Comment (Archanial) [inline code/modules/clothing/under/_under.dm]:
```suggestion
	if(has_sensor == BROKEN_SENSORS)
```

File: code/modules/clothing/under/_under.dm [hunk 4]
```diff
@@ -173,3 +173,98 @@
...
+		to_chat(usr, "The sensors have shorted out!")
+		return 0
```
Comment (Archanial) [inline code/modules/clothing/under/_under.dm]:
```suggestion
		return FALSE
```

File: code/modules/clothing/under/_under.dm [hunk 5]
```diff
@@ -173,3 +173,98 @@
...
+	if(src.has_sensor <= NO_SENSORS)
```
Comment (Archanial) [inline code/modules/clothing/under/_under.dm]:
```suggestion
	if(has_sensor <= NO_SENSORS)
```

File: code/modules/clothing/under/_under.dm [hunk 6]
```diff
@@ -173,3 +173,98 @@
...
+		to_chat(usr, "This suit does not have any sensors.")
+		return 0
```
Comment (Archanial) [inline code/modules/clothing/under/_under.dm]:
```suggestion
		return FALSE
```

File: code/modules/clothing/under/_under.dm [hunk 7]
```diff
@@ -173,3 +173,98 @@
...
+
+	var/list/modes = list("Off", "Binary vitals", "Exact vitals", "Tracking beacon")
```
Comment (Archanial) [inline code/modules/clothing/under/_under.dm]:
```suggestion
	var/list/modes = list("Off", "Binary vitals", "Exact vitals", "Tracking beacon")
```
this probably could be static

**Comment by Cenrus** [comment]
It is conflicting because the branch hasn't been updated to include the suit sensor changes on master

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
There is a bigger underlying issue here if moving the proc from 1 file to another 'fixes' the issue.
Since no code is changed and you literally just moved the proc from 1 file to another then either 1 of 2 things is happening:
 - Clothing.dm isn't being loaded from the beestation.dme
 - There is an override for verb/toggle() that breaks clothing.dm

If the second is the case, then rather than moving the verb, the second verb/toggle needs to be removed.

> this is intended, but the dumbass didn't put it under the jumpsuit code

Byond handles procs by looking at their typepath not their item.
That means that the proc
/a/b/proc/c()
will apply to the datum /a/b, no matter what file it is placed in.
Code lines can be placed in completely random files and will still work fine, and multiple different datums can be inside 1 file.
The only thing the file that code is in should affect is the order in which it is loaded.
For example:

"code/a.dm"
/obj/item/apple/proc/eat()
   world.log << "consumed"

"code/b.dm"
/obj/item/apple/proc/eat()
   world.log << "dead"

(Not sure on the order)
B will be loaded after A. Despite being in different files the typepath for the proc is the same, meaning it is both being added to /obj/item/apple. Since B.dm is loaded after A.dm, the proc in A.dm is overridden by the proc in B.dm meaning that when eat() is called on the apple, "dead" will be printed to the world log rather than consumed.

**Comment by Sylphily** [comment]
> There is a bigger underlying issue here if moving the proc from 1 file to another 'fixes' the issue.
> Since no code is changed and you literally just moved the proc from 1 file to another then either 1 of 2 things is happening:
>  - Clothing.dm isn't being loaded from the beestation.dme
>  - There is an override for verb/toggle() that breaks clothing.dm
> 
> If the second is the case, then rather than moving the verb, the second verb/toggle needs to be removed.
> 
> > this is intended, but the dumbass didn't put it under the jumpsuit code
> 
> Byond handles procs by looking at their typepath not their item.
> That means that the proc
> /a/b/proc/c()
> will apply to the datum /a/b, no matter what file it is placed in.
> Code lines can be placed in completely random files and will still work fine, and multiple different datums can be inside 1 file.
> The only thing the file that code is in should affect is the order in which it is loaded.
> For example:
> 
> "code/a.dm"
> /obj/item/apple/proc/eat()
>    world.log << "consumed"
> 
> "code/b.dm"
> /obj/item/apple/proc/eat()
>    world.log << "dead" 

> 
> (Not sure on the order)
> B will be loaded after A. Despite being in different files the typepath for the proc is the same, meaning it is both being added to /obj/item/apple. Since B.dm is loaded after A.dm, the proc in A.dm is overridden by the proc in B.dm meaning that when eat() is called on the apple, "dead" will be printed to the world log rather than consumed.
> 

I don't know What happening with the code or why it worked, I am just a clueless as when I started to fix this, I simply looked at tgs code and found clothing.dm did not have jumpsuit shit in the same file, and found it in under.dm I tried to figure out why it worked there, I finally tried moving the code to the same spot as tgs and it works .

**Comment by PowerfulBacon** [comment]
It probably worked because you are working on an outdated branch. (Hence merge conflicts).
 - Switch to master
 - Fetch
 - Merge upstream/master or origin/master whatever its called into your master
 - Push changes
 - switch to this branch
 - Update branch from master

**Comment by ike709** [comment]
At this point you need to update your master branch, make a new branch, and try this PR again.

**Pull request closed without merging**